review/get_review_ids.py

```diff
@@ -89,19 +93,31 @@ def get_vertical(current_course):
     vertical_data = set()
 
     for block_key, state in get_records(user, current_course):
-        if is_valid_problem(store, block_key, state):
-            parent = course_blocks.get_parents(block_key)[0]
-            vertical_id = parent.block_id
-            vertical_data.add(vertical_id)
+        block_key = block_key.replace(course_key=store.fill_in_run(block_key.course_key))
+        if is_valid_problem(store, block_key, state, course_blocks):
+            vertical = course_blocks.get_parents(block_key)[0]
+            sequential = course_blocks.get_parents(vertical)[0]
+            # This is in case the direct parent of a problem is not a vertical,
+            # we want to keep looking until we find the parent vertical to display.
+            # For example, you may see:
+            # sequential -> vertical -> split_test -> problem
+            # OR
+            # sequential -> vertical -> vertical -> problem
+            # OR
+            # sequential -> vertical -> conditional_block -> problem
+            while sequential.block_type != 'sequential':
```

Comment:
Why do you need to verify that the grandparent is a `sequential`?  You only care for the parent being a `vertical`, right?

Reply:
I thought this might be a better way of checking rather than only searching for verticals because of scenarios like "sequential -> vertical -> vertical -> problem". My intuition was that if we are displaying a vertical to the learner, we want the learner to see the top most vertical so by waiting until we see the sequential, we would have the top most vertical

Reply:
Ok.  As long as you don't crash or have an infinite loop in courses with weird structures (not having a sequential ancestor, for example).